csharp/device/Microsoft.Azure.Devices.Client/Common/SHA.cs

```diff
@@ -0,0 +1,211 @@
+// 
+//  Code by Elze Kool
```

Comment:
Since Elze Kool wrote it, do we have his authorization or license information on this piece of code? if not we should reach out to him first to make sure it's OK to use his code. looks like he can be found here: https://github.com/elzekool